lib/apis/moray.js

```diff
@@ -493,6 +488,34 @@ Moray.prototype._createSearch = function (params, cb) {
 };
 
 
+/**
+ * List all VMs for a given server from the VM bucket.
+ * @param  {string}     uuid uuid of the server whos vms should be fetched.
+ * @param  {Function}   cb  function of the form function(err, vms) {...}
+ */
+Moray.prototype.listVmsForServer = function listVmsForServer(uuid, cb) {
```

Comment:
@joshwilsdon The performance issue is a very good point. As per our discussions, we do not load all VMs from moray. Instead, we load all VMs that are passed in the payload of the PUT /vms request. It's still a lot, and the performance is not great but:

1. It's bounded by the number of "live" VMs that can fit on a CN.
2. By loading each VM passed as the request payload individually and [in parallel](https://github.com/joyent/sdc-vmapi/pull/11/files#diff-8491b9ad8bfb1f56e7e6c552ad1959e6R1219), and by updating them in parallel too, the performance according to my tests is actually slightly better than the previous implementation.

We also discussed a new design that would make the PUT /vms endpoint obsolete, and would get rid of the performance issues by design. I believe the latest update of this PR gives us an acceptable intermediate step towards that goal.